extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -214,20 +236,21 @@ export class AmpLightboxViewer extends AMP.BaseElement {
    * @return {!Promise}
    */
   open_(element) {
-    if (this.activeElement_ == element) {
-      return Promise.resolve();
-    }
-
-    const updateViewerPromise = this.updateViewer_(element);
     this.getViewport().enterLightboxMode();
 
     toggle(this.element, true);
     this.active_ = true;
 
+    this.updateInViewport(this.container_, true);
+    this.scheduleLayout(this.container_);
+
+    const index = Number(element.getAttribute('amp-lbv-id'));
+    this.carousel_.implementation_.showSlideWhenReady_(index);
+
     this.win.document.documentElement.addEventListener(
         'keydown', this.boundHandleKeyboardEvents_);
 
-    return updateViewerPromise;
+    return Promise.resolve();
```

Comment:
Don't need to return anything. Original code wrongly assumed `activate` which calls this method, needs to return a promise. That's not the case.